src/Integration.Vsix/SonarLintDaemon/CFamily.cs

```diff
@@ -0,0 +1,471 @@
+
+            public string RuntimeTypeInfo { get; set; }
+
+            public string BasicRuntimeChecks { get; set; }
+
+            public string OmitDefaultLibName { get; set; }
+
+            public string AdditionalOptions { get; set; }
+
+            #endregion
+
+            public Capture[] ToCaptures(string path, out string sqLanguage)
+            {
+                var p = new Capture()
+                {
+                    Executable = "cl.exe",
+                    Cwd = Path.GetDirectoryName(AbsoluteFilePath),
+                    StdErr = string.Format("{0} for {1}", ConvertPlatformToolset(PlatformToolset), ConvertPlatformName(PlatformName)),
+                    StdOut = "",
+                };
+
+                var c = new Capture()
+                {
+                    Executable = p.Executable,
+                    Cwd = p.Cwd,
+                    Env = new List<string>(),
+                    Cmd = new List<string>(),
+                };
+                c.Env.Add("INCLUDE=" + IncludeDirectories);
+                c.Cmd.Add(c.Executable);
+                Add(c.Cmd, "true".Equals(IgnoreStandardIncludePath) ? "/X" : "");
+                AddList(c.Cmd, "/I", AdditionalIncludeDirectories.Split(';'));
+                AddList(c.Cmd, "/FI", ForcedIncludeFiles.Split(';'));
+                Add(c.Cmd, "/Yu", PrecompiledHeaderFile);
+
+                Add(c.Cmd, "true".Equals(UndefineAllPreprocessorDefinitions) ? "/u" : "");
+                AddList(c.Cmd, "/D", PreprocessorDefinitions.Split(';'));
+                AddList(c.Cmd, "/U", UndefinePreprocessorDefinitions.Split(';'));
+
+                Add(c.Cmd, ConvertCompileAsAndGetSqLanguage(CompileAs, path, out sqLanguage));
+                Add(c.Cmd, ConvertCompileAsManaged(CompileAsManaged));
+                Add(c.Cmd, "true".Equals(CompileAsWinRT) ? "/ZW" : "");
+                Add(c.Cmd, "true".Equals(DisableLanguageExtensions) ? "/Za" : ""); // defines macro "__STDC__" when compiling C
+                Add(c.Cmd, "false".Equals(TreatWChar_tAsBuiltInType) ? "/Zc:wchar_t-" : ""); // undefines macros "_NATIVE_WCHAR_T_DEFINED" and "_WCHAR_T_DEFINED"
+                Add(c.Cmd, "false".Equals(ForceConformanceInForLoopScope) ? "/Zc:forScope-" : "");
+                Add(c.Cmd, "true".Equals(OpenMPSupport) ? "/openmp" : "");
+
+                Add(c.Cmd, ConvertRuntimeLibrary(RuntimeLibrary));
+                Add(c.Cmd, ConvertExceptionHandling(ExceptionHandling));
+                Add(c.Cmd, ConvertEnableEnhancedInstructionSet(EnableEnhancedInstructionSet));
+                Add(c.Cmd, "true".Equals(OmitDefaultLibName) ? "/Zl" : ""); // defines macro "_VC_NODEFAULTLIB"
+                Add(c.Cmd, "false".Equals(RuntimeTypeInfo) ? "/GR-" : ""); // undefines macro "_CPPRTTI"
+                Add(c.Cmd, ConvertBasicRuntimeChecks(BasicRuntimeChecks));
+
+                // TODO Q: what if it contains space in double quotes?
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)